docs/taskbackground.rst

```diff
@@ -35,23 +65,126 @@ The tree of the above event looks something like
         │   └── c~
         └── b~
         
-Top Mass Reconstruction
+To ensure that we retain the highest possible purity of :math:`t\bar{t}` events in the signal region, we make the following cuts:
+
+* Events must contain exactly one lepton, which must have :math:`p_T>30` GeV, :math:`|\eta|<2.1`, and ``sip3d<4`` (significance of 3d impact parameter)
+* Jets must have :math:`p_T>30` GeV, :math:`|\eta|>2.4`, and ``isTightLeptonVeto``
+* Events must have at least four jets
+* Events must have exactly one :math:`b`-tagged jet
+
+There are a few more details about the cuts, which one can read about in **AGC Versions** in the **Cuts** section.
+
+
+3. Weighting
+---------------------------------------------------------------
+The above-described algorithm assumes that data samples generated using different channels come into the histogram with equivalent weights. 
+This is not the case, as the relative number of events in each sample does not reflect the rate at which that process occurs in our detected events. 
+So we need to account for the cross-section when processing events. This is the formula for calculating weights:
+
+.. math::
+    
+    w_i = {{\sigma}_i L \over N_i}
+
+where :math:`i` represents the partial interaction channel, :math:`\sigma` is the partial cross-section, :math:`L` is the luminosity, and :math:`N_i` is the total number of events in the data sample.
+
+The luminosity for this dataset is :math:`L=3378\;pb^{-1}`.
+
+The cross-section values used are listed below (obtained from `this site <https://twiki.cern.ch/twiki/bin/view/LHCPhysics/LHCTopWG>`_):
+
+.. list-table:: Cross-section values
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Process
+     - Cross-section [pb]
+   * - ttbar
+     - 729.84
+   * - single_top_s_chan
+     - 6.839
+   * - single_top_t_chan
+     - 214.2
+   * - single_top_tW
+     - 79.3
+   * - wjets
+     - 61526.7 * 0.253
+
+4. Top Mass Reconstruction
 ---------------------------------------------------------------
-To measure the :math:`t\bar{t}` cross-section, we use an observable that approximately reconstructs the top quark mass. We do this in two different ways. The first uses no machine learning. Within an event, the trijet system with the highest transverse momentum (:math:`p_T`) is selected. We then calculate the combined mass of these three jets.
+To measure the :math:`t\bar{t}` cross-section, we use an observable that approximately reconstructs the top quark mass. 
+This is done using the following steps:
 
-Machine Learning Component
+#. Filter the events using the criteria explained above (including requiring at least four jets and exactly one lepton)
+#. Calculate all possible combinations of three jets from all jets in each event (utilizes ``ak.combinations``)
```

Comment:
Let's try to distinguish the implementation and the task a bit more clearly here: the bracket can mention that one way of accomplishing this (with `awkward`) is this function, but we should make clear that this is not the unique solution.